sdk/core/azure-core/inc/azure/core/azure_assert.hpp

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+/**
+ * @file
+ * @brief Deprecated assert macros.
```

Comment:
Maybe we should be more explicit in the comment.

```suggestion
 * @brief Deprecated assert macros. It is only temporarily made available for compatibility and should NOT be used by any callers outside of the SDK.
```